apps/asap-server/src/handlers/teams/index-handler.ts

```diff
@@ -0,0 +1,60 @@
+import { EventBridgeEvent } from 'aws-lambda';
+import algoliasearch, { SearchClient } from 'algoliasearch';
+import { TeamResponse } from '@asap-hub/model';
+import { TeamsEventType } from '../webhooks/webhook-teams';
+import Team, { TeamController } from '../../controllers/teams';
+import {
+  algoliaAppId,
+  algoliaIndexApiKey,
+  algoliaResearchOutputIndex,
+} from '../../config';
+import logger from '../../utils/logger';
+
+export const indexHandlerFactory =
+  (
+    teamController: TeamController,
+    algoliaClient: SearchClient,
+  ): ((
+    event: EventBridgeEvent<TeamsEventType, SquidexWebhookTeamPayload>,
+  ) => Promise<void>) =>
+  async (
+    event: EventBridgeEvent<TeamsEventType, SquidexWebhookTeamPayload>,
+  ): Promise<void> => {
+    const team = await teamController.fetchById(event.detail.payload.id);
+
+    logger.info(`Fetched team with id ${team.id}`);
+
+    await indexResearchOutputs(team, algoliaClient);
```

Comment:
I see you tested the handler without mocking this so it's not really super relevant but... 

you could make the `indexResearchOutput` a function that takes a `ResearchOutputResponse` type and use it here and in [this handler](https://github.com/yldio/asap-hub/blob/master/apps/asap-server/src/handlers/research-output/index-handler.ts). Technically that would abstract away the algolia implementation and would enable you to mock it (which may become handy should you need to test some edge cases here).

Reply:
one more thing - your `indexResearchOutputs` takes a dependency that it inherits from `indexHandlerFactory` - thats not a good pattern, in general, [see here](https://en.wikipedia.org/wiki/Dependency_inversion_principle)

`indexHandlerFactory` is not even interested in `algoliaClient`, so realistically it should instead take a `indexResearchOutput` dependency instead and not care about what that function does.

Reply:
finally :D you could just leave it as is because your `indexResearchOutputs` is just a wrapper around some logic - it's just strange to see that you're having to pass down the algoliaClient - I guess since your `indexHandlerFactory` is a factory you could add the init phase to it (before returning the function) and define your `indexResearchOutputs` there - that way `algoliaClient` will already be available in the scope